src/window/src/lib.rs

```diff
@@ -21,54 +35,93 @@ pub struct Size {
     pub height: u32,
 }
 
+/// Convinience conversion between an array of size two, and a size structure.
 impl From<[u32; 2]> for Size {
     #[inline(always)]
     fn from(value: [u32; 2]) -> Size {
         Size { width: value[0], height: value[1] }
     }
 }
 
+/// Convinience conversion between a tuple of (width,height), and a size structure.
 impl From<(u32, u32)> for Size {
     #[inline(always)]
     fn from(value: (u32, u32)) -> Size {
         Size { width: value.0, height: value.1 }
     }
 }
 
-/// Builds window from window settings.
+/// This trait indicates that a window can be built from a WindowSettings object.
 pub trait BuildFromWindowSettings: Sized {
-    /// Builds window from window settings.
+    /// Builds the window from a window settings object.
+    ///
+    /// # Errors
+    ///
+    /// See your backend's documentation for details about what kind of
+    /// error strings can be returned, and the conditions for error.
     fn build_from_window_settings(settings: WindowSettings)
     -> Result<Self, String>;
 }
 
-/// Required to use the event loop.
+/// Trait for piston internals to interface with a window backend.
+///
+/// This trait defines all the behavior needed for making an event loop.
+/// Event loop definitions can be found in [`piston::event_loop`]
+/// (../event_loop/index.html).
+/// An example of a working event loop can be found in the Piston-Tutorials
+/// repository, under getting-started, or in the event loop examples.
+/// The window trait methods are private; they're mostly for use in the abstract
```

Comment:
Traits don't have private methods. There is no `pub` because the methods are public by default.